doc/manual/src/language/string-context.md

```diff
@@ -0,0 +1,144 @@
+Explicitly discarding the string context and then expecting it helps ensure that string context elements are not lost by mistake.
+
+## Constructing string contexts
+
+[`builtins.appendContext`] will create copy of a string but with additional string context elements.
+The context is specified explicitly by an [attribute set] in the format that [`builtins.hasContext`] produces.
+We there can create strings arbitrary contexts in 3 steps:
+
+1. Create strings with the string context elements we want
+2. Dump their contexts with [`builtins.getContext`]
+3. Combine them together with a base string and repeated [`builtins.appendContext`] calls.
+
+The remainder of this section will focus on step 1: making strings with individual string context elements on which to apply `builtins.getContext`.
+
+[`builtins.appendContext`]: ./builtins.md#builtins-appendContext
+
+## Constant string context elements
+
+A constant string context element is just a constant [derived path];
+a constant derived path is just a [store path].
+We therefore want to use [`builtins.storePath`] to create a string with a single constant string context element:
+
+> **Example**
+>
+> ```nix
+> builtins.getContext (builtins.storePath "/nix/store/wkhdf9jinag5750mqlax6z2zbwhqb76n-hello-2.10")
+> ```
+> evaluates to
+> ```nix
+> {
+>   "/nix/store/wkhdf9jinag5750mqlax6z2zbwhqb76n-hello-2.10" = {
+>     path = true;
+>   };
+> }
+> ```
+
+[derived path]: @docroot@/glossary.md#gloss-derived-path
+[store path]: @docroot@/glossary.md#gloss-store-path
+[`builtins.storePath`]: ./builtins.md#builtins-storePath
+
+## Output string context elements
+
+The best way to illustrate this with a builtin function that is still experimental: [`builtins.ouputOf`].
+This example will *not* work the stable Nix!
+
+> **Example**
+>
+> ```nix
+> builtins.getContext
+>   (builtins.outputOf
+>     (builtins.storePath "/nix/store/fvchh9cvcr7kdla6n860hshchsba305w-hello-2.12.drv")
+>     "out")
+> ```
+> evaluates to
+> ```nix
+> {
+>   "/nix/store/fvchh9cvcr7kdla6n860hshchsba305w-hello-2.12.drv" = {
+>     outputs = [ "out" ];
+>   };
+> }
+> ```
+
+[`builtins.outputOf`]: ./builtins.md#builtins-outputOf
+
+## "Derivation deep" string context elements
+
+These best way to illustrate this is with [`builtins.addDrvOutputDependencies`].
+We take a regular constant string context element pointing to a derivation, and transform it unto a "Derivation deep" string context element.
+
+> **Example**
+>
+> ```nix
+> builtins.getContext
+>   (builtins.addDrvOutputDependencies
+>     (builtins.storePath "/nix/store/fvchh9cvcr7kdla6n860hshchsba305w-hello-2.12.drv"))
+> ```
+> evaluates to
+> ```nix
+> {
+>   "/nix/store/fvchh9cvcr7kdla6n860hshchsba305w-hello-2.12.drv" = {
+>     allOutputs = true;
+>   };
+> }
+> ```
+
+[`builtins.unsafeDiscardOutputDependency`] does this the opposite of [`builtins.addDrvOutputDependencies`], but is not prefered because it "weakens" rather than "strengens" the string context.
```

Comment:
```suggestion
[`builtins.unsafeDiscardOutputDependency`] does the opposite of [`builtins.addDrvOutputDependencies`], but is not preferred because it "weakens" rather than "strengthens" the string context.
```

Reply:
```suggestion
[`builtins.unsafeDiscardOutputDependency`] does this the opposite of [`builtins.addDrvOutputDependencies`], but is marked unsafe because it "forgets" the dependencies on the outputs, whereas Nix normally tracks all dependencies consistently.
```

probably don't need the next sentence if you apply this.